compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -916,20 +923,31 @@ impl<'interner> Monomorphizer<'interner> {
                     ast::Expression::Ident(ident)
                 }
             },
-            DefinitionKind::GenericType(type_variable) => {
+            DefinitionKind::GenericType(type_variable, numeric_typ) => {
```

Comment:
It seems like we should rename this to `DefinitionKind::NumericGeneric` as it's not really a type and they're just carrying the value of the numeric generic. Shouldn't be done in this PR however.

Reply:
https://github.com/noir-lang/noir/issues/6181